library/proc_macro/src/lib.rs

```diff
@@ -1233,6 +1233,7 @@ pub mod tracked_env {
     /// Besides the dependency tracking this function should be equivalent to `env::var` from the
     /// standard library, except that the argument must be UTF-8.
     #[unstable(feature = "proc_macro_tracked_env", issue = "74690")]
+    #[cfg_attr(not(bootstrap), allow(duplicate_bounds))]
     pub fn var<K: AsRef<OsStr> + AsRef<str>>(key: K) -> Result<String, VarError> {
```

Comment:
This seems to be a false positive from the old clippy lint as well:
```rust
#![deny(clippy::pedantic)]
#![deny(clippy::pedantic)]

fn foo<T: AsRef<str>>()
where
    T: AsRef<String>,
{
}
```
```
error: this trait bound is already specified in the where clause
 --> src/lib.rs:3:11
  |
3 | fn foo<T: AsRef<str>>()
  |           ^^^^^^^^^^
  |
```
I guess I would have to compare `Path::segments` instead of `Res` to handle this case?